sys/net/gnrc/routing/rpl/gnrc_rpl_control_messages.c

```diff
@@ -184,8 +184,13 @@ gnrc_pktsnip_t *_dio_prefix_info_build(gnrc_pktsnip_t *pkt, gnrc_rpl_dodag_t *do
     if (_get_pl_entry(dodag->iface, &dodag->dodag_id, prefix_info->prefix_len,
                       &ple)) {
         uint32_t now = (xtimer_now_usec64() / US_PER_MS) & UINT32_MAX;
-        prefix_info->valid_lifetime = byteorder_htonl((ple.valid_until - now) / MS_PER_SEC);
-        prefix_info->pref_lifetime = byteorder_htonl((ple.pref_until - now) / MS_PER_SEC);
+        uint32_t valid_ltime = (ple.valid_until < UINT32_MAX) ?
+                               ple.valid_until / MS_PER_SEC : UINT32_MAX;
+        uint32_t pref_ltime = (ple.pref_until < UINT32_MAX) ?
+                              ple.pref_until / MS_PER_SEC : UINT32_MAX;
```

Comment:
I think this should be:
```
uint32_t pref_ltime = (ple.pref_until < UINT32_MAX) ?
                      (ple.pref_until - now) / MS_PER_SEC : UINT32_MAX;
```

Same issue with the valid_until.

Reply:
Gnarrfff I need the weekend, I need this to be over ^^

Reply:
Fixed